src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -573,6 +573,12 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
               else
                 new ApplyToUntyped(tree, fun1, funRef, proto, pt)(argCtx(tree))
             convertNewGenericArray(ConstFold(app.result))
+          case mt: MethodType =>
```

Comment:
I would squash this commit and the next one in the third commit, no need to record everything that was tried.